### PR TITLE
Added simple unit tests in tests CI

### PR DESCRIPTION
**Pull request by andreped**
Related to issue https://github.com/awsaf49/gcvit-tf/issues/13.

For every commit, PR, and dispatch, unit tests are now being performed for selected operating systems, python versions, and tensorflow versions. More/different versions can be added if of interest.

The tests added are very simple and only check the two use cases demonstrated in the README. Future tests can be added if necessary. It makes more sense to me to add future unit tests in seperate PRs. This PR only adds the core unit tests through GitHub Actions feature.

Note that predictions may differ slightly between `tensorflow` versions, as was observed in the unit tests. However, with `decimal=4` the predictions are sufficiently close.

I also had to bump the version to `v1.1.3` due to [this line](https://github.com/andreped/gcvit-tf/blob/ci/.github/workflows/tests.yml#L53) in the CI, where the wheel on PyPI was used instead of the precompiled one for whatever reason.

### CODE REVIEW

File: tests/test_gcvit.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        chelsea(), mode="torch"
+    )
+    img = tf.image.resize(img, (224, 224))[
+        None,
+    ]
+
+    model = GCViTTiny(pretrain=True)
+
+    pred = model(img).numpy()
+    result = tf.keras.applications.imagenet_utils.decode_predictions(pred)[0]
```
Comment (awsaf49) [inline tests/test_gcvit.py:20]: Can you also add arg, `top=5`

File: tests/test_gcvit.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    expected_result = [
+        ("n02124075", "Egyptian_cat", 0.7402193),
+        ("n02123045", "tabby", 0.07757583),
+        ("n02123159", "tiger_cat", 0.07006007),
+        ("n02127052", "lynx", 0.0043750308),
+        ("n04040759", "radiator", 0.00091264246),
+    ]
+    for res, exp in zip(result, expected_result):
+        assert res[1] == exp[1]
+        assert np.around(res[2], decimals=4) == np.float32(
```
Comment (awsaf49) [inline tests/test_gcvit.py:36]: Can you try `np.allclose` ? I think it will reduce the line size

File: tests/test_gcvit.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import numpy as np
+import tensorflow as tf
+from skimage.data import chelsea
+
+from gcvit import GCViTTiny
```
Comment (awsaf49) [inline tests/test_gcvit.py:5]: To reduce CI time we can try smaller models like `GCViTXXTiny`. Output feature or feature map may change

File: tests/test_gcvit.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+def test_inference():
+    # load test image, resize, and create batch
+    img = tf.keras.applications.imagenet_utils.preprocess_input(
+        chelsea(), mode="torch"
+    )
+    img = tf.image.resize(img, (224, 224))[
+        None,
+    ]
+
+    model = GCViTTiny(pretrain=True)
```
Comment (awsaf49) [inline tests/test_gcvit.py:17]:
```suggestion
    model = GCViTXXTiny(pretrain=True)
```

File: tests/test_gcvit.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+def test_feature_extraction():
+    # load test image, resize, and create batch
+    img = tf.keras.applications.imagenet_utils.preprocess_input(
+        chelsea(), mode="torch"
+    )
+    img = tf.image.resize(img, (224, 224))[
+        None,
+    ]
+
+    model = GCViTTiny(pretrain=True)
```
Comment (awsaf49) [inline tests/test_gcvit.py:50]:
```suggestion
    model = GCViTXXTiny(pretrain=True)
```

**Comment by andreped** [comment]
Will likely need to update the assertion where we compare prediction values, as we have changed network. Can pull and update locally as it should go faster.

**Comment by andreped** [comment]
I made the necessary modifications.

File: gcvit/version.py
Comment (andreped) [inline gcvit/version.py:1]: @awsaf49 Note that you might want to revert this back before making a new release, to avoid skipping a patch version.

**Pull request merged**